Sofa/Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/TetrahedronHyperelasticityFEMForceField.inl

```diff
@@ -220,188 +237,202 @@ void TetrahedronHyperelasticityFEMForceField<DataTypes>::createTetrahedronRestIn
     // store the point position
 
     for (j = 0; j < 4; ++j)
-        point[j] = (restPosition)[t[j]];
+        point[j] = restPosition[t[j]];
     /// compute 6 times the rest volume
     volume = dot(cross(point[2] - point[0], point[3] - point[0]), point[1] - point[0]);
     /// store the rest volume
     tinfo.m_volScale = (Real)(1.0 / volume);
     tinfo.m_restVolume = fabs(volume / 6);
     // store shape vectors at the rest configuration
-    for (j = 0; j < 4; ++j) {
+    for (j = 0; j < 4; ++j)
+    {
         if (!(j % 2))
-            tinfo.m_shapeVector[j] = -cross(point[(j + 2) % 4] - point[(j + 1) % 4], point[(j + 3) % 4] - point[(j + 1) % 4]) / volume;
+            tinfo.m_shapeVector[j] = -cross(point[(j + 2) % 4] - point[(j + 1) % 4],
+                                            point[(j + 3) % 4] - point[(j + 1) % 4]) / volume;
         else
-            tinfo.m_shapeVector[j] = cross(point[(j + 2) % 4] - point[(j + 1) % 4], point[(j + 3) % 4] - point[(j + 1) % 4]) / volume;;
+            tinfo.m_shapeVector[j] = cross(point[(j + 2) % 4] - point[(j + 1) % 4],
+                                           point[(j + 3) % 4] - point[(j + 1) % 4]) / volume;;
     }
 
-
-    for (j = 0; j < 6; ++j) {
+    for (j = 0; j < 6; ++j)
+    {
         Edge e = m_topology->getLocalEdgesInTetrahedron(j);
         int k = e[0];
         //int l=e[1];
-        if (edgeArray[te[j]][0] != t[k]) {
+        if (edgeArray[te[j]][0] != t[k])
+        {
             k = e[1];
```

Comment:
This code is weird: `k` is never used. I am not sure this loop is useful, or it is buggy.

Reply:
Indeed buggy and doing nothing... 
maybe it was used to send warning? Did you check the history?